apps/server/src/modules/task/controller/api-test/task.api.spec.ts

```diff
@@ -66,724 +83,881 @@ class API {
 
-				await em.persistAndFlush([teacher, task]);
-				em.clear();
+				const { data } = response.body as TaskListResponse;
+				expect(response.statusCode).toBe(200);
+				expect(data[0].id).toContain(task.id);
+			});
 
-				currentUser = mapUserToCurrentUser(teacher);
-				const params = { courseId: course.id };
+			it('student does not find finished tasks, if tasks have users assigned, but himself is not assigned', async () => {
+				const { notAssignedStudent } = await setup();
 
-				const response = await request(app.getHttpServer())
-					.post(`/tasks/${task.id}/copy`)
-					.set('Authorization', 'jwt')
-					.send(params);
+				const response = await apiRequest.get('/finished', notAssignedStudent.account);
 
-				expect(response.status).toEqual(201);
+				const { total } = response.body as TaskListResponse;
+				expect(response.statusCode).toBe(200);
+				expect(total).toBe(0);
 			});
+		});
+	});
 
-			it('should duplicate a task avoiding name collisions', async () => {
-				const teacher = setup();
-				const course = courseFactory.build({
-					teachers: [teacher],
-				});
-				const originalTask = taskFactory.build({ creator: teacher, course, name: 'Addition' });
-				const task2 = taskFactory.build({ creator: teacher, course, name: 'Addition (1)' });
-				const task3 = taskFactory.build({ creator: teacher, course, name: 'Addition (3)' });
-
-				await em.persistAndFlush([teacher, originalTask, task2, task3]);
-				em.clear();
-
-				currentUser = mapUserToCurrentUser(teacher);
+	// TODO: refactor
+	describe('When task-card feature is enabled', () => {
+		// eslint-disable-next-line @typescript-eslint/no-shadow
+		let app: INestApplication;
+		// eslint-disable-next-line @typescript-eslint/no-shadow
+		let em: EntityManager;
+		let currentUser: ICurrentUser;
 
-				const result = await api.copyTask(originalTask.id, course.id);
-				expect(result.status).toEqual(201);
-				expect(result.copyStatus?.title).toEqual('Addition (2)');
+		const setup = (permission) => {
+			const roles = roleFactory.buildList(1, {
+				permissions: [permission],
 			});
+			const user = userFactory.build({ roles });
 
-			it('should avoid name collisions when copying the same task twice', async () => {
-				const teacher = setup();
-				const course = courseFactory.build({
-					teachers: [teacher],
-				});
-				const originalTask = taskFactory.build({ creator: teacher, course, name: 'Addition' });
-
-				await em.persistAndFlush([teacher, course, originalTask]);
-				em.clear();
+			return user;
+		};
 
-				currentUser = mapUserToCurrentUser(teacher);
+		beforeAll(async () => {
+			const module: TestingModule = await Test.createTestingModule({
+				imports: [ServerTestModule],
+			})
+				.overrideGuard(JwtAuthGuard)
+				.useValue({
+					canActivate(context: ExecutionContext) {
+						const req: Request = context.switchToHttp().getRequest();
+						req.user = currentUser;
+						return true;
+					},
+				})
+				.compile();
```

Comment:
Why do you have beforeAll ?
Maybe you move this to other file

Reply:
These are tests which are not refactored yet.
The PR is split up.